mobile/native/src/orderbook.rs

```diff
@@ -38,6 +39,9 @@ pub fn subscribe(secret_key: SecretKey) -> Result<()> {
             Signature { pubkey, signature }
         };
 
+        // Consider using a HashMap instead to optimize the lookup for removal/update
```

Comment:
🙃 I'd action that TODO now - we can still easily extract the `values` of the hashmap as `Vec` if that's more convenient for `best_current_price`. Don't mind much though.

Reply:
I will amend the `best_current_price` to also take the hashmap, because it's easier - but after I have it end-to-end working, otherwise it's just an optimisation :-)

note: even `AllOrders` would be best as HashMap, because then we can specify stricter contract (ie. no duplicates!).